### PR TITLE
BUGFIX: SETI-384 Zuul - handle ping events

### PR DESCRIPTION
**Pull request by janpriessnitz**
Added tests for Github ping event

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -734,6 +734,19 @@ def getPushEvent(self, project, branch, old_rev=None, new_rev=None):
         }
         return (name, data)
 
+    def getPingEvent(self, project='whatever', hook_id=1234567, events=['push']):
+        name = 'ping'
+        data = {
```
Comment (hrubi) [inline tests/base.py]: Mocking only empty data `{}` should be enough, as you don't process it in the `_event_ping` at all. What was your intention for having something in the data?

If you remove this, you can also remove the `getPingEvent` parameters.

File: tests/test_github.py [hunk 2]
```diff
@@ -495,3 +495,8 @@ def test_failed_change_at_head(self):
         self.assertNotIn('merge', A.labels)
         self.assertNotIn('merge', B.labels)
         self.assertNotIn('merge', C.labels)
+
+    def test_ping_event(self):
+        ret = self.fake_github.emitEvent(self.fake_github.getPingEvent())
+        self.waitUntilSettled();
```
Comment (hrubi) [inline tests/test_github.py]: You don't need the `waitUntilSettled`, that's for waiting on the zuul's queues. The ping test involves only contacting the webapp, so there are no such things involved.

**Comment by janpriessnitz** [comment]
Sorry, forgot to push new version. Thanks very much for the review :D Should be OK now.

**Comment by hrubi** [comment]
Great. Please squash the commits into one, no need to having them separate in this case.

**Comment by janpriessnitz** [comment]
Done

**Pull request merged**